lib/modules/manager/gleam/artifacts.ts

```diff
@@ -49,7 +49,13 @@ export async function updateArtifacts(
       ],
     };
 
-    await exec('gleam deps download', execOptions);
+    // `gleam deps update` with no packages rebuilds the lock file
+    const packagesToUpdate = isLockFileMaintenance
+      ? []
+      : updatedDeps.map((dep) => dep.depName).filter(Boolean);
```

Comment:
I appreciate the review! This is a good suggested change - if I push into the same branch, I think I'll need to open a new PR though since this one has been closed?

Reply:
yes, you need a new pr

Reply:
Changes applied in #32962